frontend/src/components/List/Column/StateCacheColumn.tsx

```diff
@@ -0,0 +1,46 @@
+import * as React from 'react';
+import { Types, Maybe, timestamp } from '../../../common';
+import { Column, BANDWIDTH_SCALE } from './';
+import { Node } from '../../../state';
+import { Sparkline } from '../../';
+import icon from '../../../icons/git-branch.svg';
+
+export class StateCacheColumn extends React.Component<Column.Props, {}> {
+  public static readonly label = 'State Cache Size';
+  public static readonly icon = icon;
+  public static readonly width = 40;
+  public static readonly setting = 'stateCacheSize';
+  public static readonly sortBy = ({ stateCacheSize }: Node) =>
+    stateCacheSize.length < 3 ? 0 : stateCacheSize[stateCacheSize.length - 1];
+
+  private data: Array<number> = [];
+
+  public shouldComponentUpdate(nextProps: Column.Props) {
+    // Diffing by ref, as data is an immutable array
+    return this.data !== nextProps.node.stateCacheSize;
```

Comment:
There is `PureComponent` that provides a default `shouldComponentUpdate` implementation which diffs all passed props, but in this case it wouldn't know it needs to check on a property of a prop (it would only check if `this.props.node !== nextProps.node`, which isn't what we want).

The data array itself is more of the artefact of how BE sends updates here: it will send a new array of data points when something changes.